palace/linalg/errorestimator.cpp

```diff
@@ -91,260 +77,267 @@ FluxProjector<SmoothFluxFiniteElementCollection>::FluxProjector(
   double normalization = 0.0;
-
-  // Coefficient for computing the discontinuous flux., i.e. (Vᵢ, (ϵ ∇ ϕ)ᵢ).
-  GradFluxCoefficient coef(field_gf, mat_op);
-  for (int c = 0; c < sdim; c++)
+  for (int e = 0; e < mesh.GetNE(); e++)
   {
-    coef.SetComponent(c);
-    {
-      // Given the RHS vector of non-smooth flux component, compute fᵢ = M⁻¹ f̂ᵢ.
-      Mpi::Print("Computing smooth flux approximation of component {}\n", c);
-      BlockTimer bt0(Timer::ESTSOLVE);
-      mfem::LinearForm rhs(&h1_fespace);
-      rhs.AddDomainIntegrator(new mfem::DomainLFIntegrator(coef));
-      rhs.UseFastAssembly(false);
-      rhs.Assemble();
-      h1_fespace.GetProlongationMatrix()->MultTranspose(rhs, flux_rhs);
-      smooth_projector.Mult(flux_rhs, smooth_flux);
-    }
-    smooth_flux_gf[c].SetFromTrueDofs(smooth_flux);
-    smooth_flux_gf[c].ExchangeFaceNbrData();
-  }
-
-  Vector coef_eval(3);
-  for (int e = 0; e < h1_fespace.GetNE(); e++)
-  {
-    auto &T = *h1_fespace.GetElementTransformation(e);
-    // integration order 2p + q
-    const auto &ir = mfem::IntRules.Get(T.GetGeometryType(),
-                                        2 * h1_fespace.GetFE(e)->GetOrder() + T.Order());
-    for (const auto &ip : ir)
+    const mfem::FiniteElement &fe = *h1d_fespace.GetFE(e);
+    mfem::ElementTransformation &T = *mesh.GetElementTransformation(e);
+    const int q_order = 2 * fe.GetOrder() + T.OrderW();
+    const mfem::IntegrationRule &ir = mfem::IntRules.Get(T.GetGeometryType(), q_order);
+    double err = 0.0;
+    for (int i = 0; i < ir.GetNPoints(); i++)
     {
+      // XX TODO: For now the flux is just ∇U since the coefficient support for
+      // matrix-valued
+      //          permittivity is not yet there (coming soon).
+      const mfem::IntegrationPoint &ip = ir.IntPoint(i);
       T.SetIntPoint(&ip);
-      coef.Eval(coef_eval, T, ip);
-      for (int c = 0; c < sdim; c++)
-      {
-        coef.SetComponent(c);
-        double smooth_val = smooth_flux_gf[c].GetValue(e, ip);
-        const double w_i = ip.weight * T.Weight();
-        estimates[e] += w_i * std::pow(smooth_val - coef_eval(c), 2.0);
-        normalization += w_i * std::pow(coef_eval(c), 2.0);
-      }
+      const double w = ip.weight * T.Weight();
+      U_gf.GetGradient(T, V_smooth);
+      // mat_op.GetPermittivityReal(T.Attribute).Mult(V_smooth, V_ip);
+      V_ip = V_smooth;
+      F_gf.GetVectorValue(T, ip, V_smooth);
+      V_smooth -= V_ip;
+      err += w * (V_smooth * V_smooth);
+      normalization += w * (V_ip * V_ip);
     }
-    estimates[e] = std::sqrt(estimates[e]);
+    estimates[e] = std::sqrt(err);
   }
 
-  Mpi::GlobalSum(1, &normalization, h1_fespace.GetComm());
+  // Finalize the element-wise error estimates.
+  Mpi::GlobalSum(1, &normalization, mesh.GetComm());
   normalization = std::sqrt(normalization);
-  if (normalization > 0)
+  if (normalization > 0.0)
   {
-    estimates /= normalization;
+    estimates *= 1.0 / normalization;
```

Comment:
If there's a meaningful difference here, this feels like it should be reported to mfem as a bug.

Reply:
I'm not sure I follow. I have to admit I made this change just because it feels better to me to ensure we are taking care of the 1/x on our end, but looking at the implementation of `/=` it looks like it's identical. Am I missing something?